src/Raven.Server/Rachis/TimeoutEvent.cs

```diff
@@ -117,27 +117,32 @@ public void DisableTimeout()
             }
         }
 
+        private int _inProgress;
+
         public void ExecuteTimeoutBehavior()
         {
-            lock (this)
-            {
-                if (Disable)
```

Comment:
Shouldn't we take into account that it might be disabled?

Reply:
yes, was too eager on removing the `lock` :)